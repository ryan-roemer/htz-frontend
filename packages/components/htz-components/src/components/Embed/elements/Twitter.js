/* *************************************************************** *
 * This element accepts these inputTemplates:
  [
    com.polobase.TwitterEmbed,
  ]
 * *************************************************************** */

/* eslint-disable react/no-danger */
/* globals twttr */
import React from 'react';
import PropTypes from 'prop-types';
import { createComponent, } from 'react-fela';

import { appendScript, } from '../../../utils/scriptTools';

const twitterWrapper = ({ embedType, }) => {
  let height;

  switch (embedType) {
    case 'collection timeline':
    case 'moment':
      break;

    case 'search timeline':
      height = 'auto';
      break;

    case 'video':
      height = 'auto';
      break;

    case 'single tweet':
      height = 'auto';
      break;

    default:
  }
  return {
    overflow: 'auto',
    height: height || '400px',
    maxHeight: '800px',
    position: 'relative',
    margin: '0 auto',
    '& .twitter-tweet': {
      marginBottom: '0!important',
    },
  };
};

const TwitterWrapper = createComponent(twitterWrapper, 'figure');

export default class Twitter extends React.Component {
  static propTypes = {
    /**
     * The type of this twitter element
     * ('single tweet', 'video', 'search', etc).
     */
    embedType: PropTypes.string.isRequired,
    /**
     * Twitter's HTML tag (supplied by Twitter).
     */
    content: PropTypes.string.isRequired,
    /**
     * A function to be called when the element finishes to load.
     */
    onLoadCallback: PropTypes.func,
  };

  static defaultProps = {
    onLoadCallback: null,
  };

  componentDidMount() {
    const src = '//platform.twitter.com/widgets.js';
    const async = true;
    const id = 'twitter-js';
    /* istanbul ignore next */
    appendScript(src, id, async, this.initScript, this.updateScript);
  }

  initScript = () => {
    twttr.events.bind(
      'rendered', event => {
        console.log('twitter embed is loaded');
        this.props.onLoadCallback ? this.props.onLoadCallback() : '';
      }
    );
  };

  updateScript = () => {
    twttr.widgets.load();
  };

  render() {
    const { content, } = this.props;

    return (
      <TwitterWrapper embedType={this.props.embedType}>
        <div dangerouslySetInnerHTML={{ __html: content, }} />
      </TwitterWrapper>
    );
  }
}
