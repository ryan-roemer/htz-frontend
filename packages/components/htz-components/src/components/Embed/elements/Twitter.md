<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
**Table of Contents**

- [Code Example](#Twitter-component)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

### Twitter component

An example for Twitter ('single tweet') Embed.

```jsx static
<Twitter
  embedType = "single tweet"
  content = {`
  <blockquote class="twitter-tweet" data-lang="en">
    <p lang="en" dir="ltr">
      <a href="https://twitter.com/hashtag/Netflix?src=hash">#Netflix</a>&#39;s 
      <a href="https://twitter.com/hashtag/BoJackHorseman?src=hash">#BoJackHorseman</a> 
      finally has a season 4 premiere date 
      <a href="https://t.co/xt08ol2mN2">https://t.co/xt08ol2mN2</a> 
      <a href="https://t.co/IYI8pq05rc">pic.twitter.com/IYI8pq05rc</a>
    </p>&mdash; UPROXX (@UPROXX)
    <a href="https://twitter.com/UPROXX/status/884416188280864768">July 10, 2017</a>
  </blockquote>`}
  caption = "single tweet"
  credit = "Twitter"
/>
```
