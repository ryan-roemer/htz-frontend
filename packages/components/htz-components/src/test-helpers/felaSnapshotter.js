import React from 'react';
import createRenderer from '../components/StyleProvider/createRenderer';
import StyleProvider from '../components/StyleProvider/StyleProvider';
import createFelaSnapshotter from './createFelaSnapshotter';

const testRenderer = createRenderer({ isRtl: true, });
const felaSnapshotter = createFelaSnapshotter(StyleProvider, testRenderer);

export default felaSnapshotter;
