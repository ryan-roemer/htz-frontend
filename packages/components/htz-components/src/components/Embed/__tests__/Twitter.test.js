import React from 'react';
import felaSnapshotter from '../../../test-helpers/felaSnapshotter';

import Twitter from '../elements/Twitter';

it('Twitter single tweet ', () => {
  const snapshot = felaSnapshotter(
    <Twitter
      content="<blockquote class=\&quot;twitter-tweet\&quot; data-lang=\&quot;en\&quot;><p lang=\&quot;en\&quot; dir=\&quot;ltr\&quot;><a href=\&quot;https://twitter.com/hashtag/Netflix?src=hash\&quot;>#Netflix</a>&#39;s <a href=\&quot;https://twitter.com/hashtag/BoJackHorseman?src=hash\&quot;>#BoJackHorseman</a> finally has a season 4 premiere date <a href=\&quot;https://t.co/xt08ol2mN2\&quot;>https://t.co/xt08ol2mN2</a> <a href=\&quot;https://t.co/IYI8pq05rc\&quot;>pic.twitter.com/IYI8pq05rc</a></p>&mdash; UPROXX (@UPROXX) <a href=\&quot;https://twitter.com/UPROXX/status/884416188280864768\&quot;>July 10, 2017</a></blockquote>"
      caption="tweet"
      credit="tweet"
      embedType="single tweet"
      elementType="embedElement"
      inputTemplate="com.polobase.TwitterEmbed"
      contentId="7.3909"
      contentName="tweet"
    />
  );
  expect(snapshot).toMatchSnapshot();
});

it('Twitter grid ', () => {
  const snapshot = felaSnapshotter(
    <Twitter
      content="<a class=\&quot;twitter-grid\&quot; data-partner=\&quot;tweetdeck\&quot; href=\&quot;https://twitter.com/woodenecho/timelines/890172042527592448\&quot;>Testing collections</a>"
      caption="grid Caption"
      credit="grid"
      embedType="collection timeline"
      elementType="embedElement"
      inputTemplate="com.polobase.TwitterEmbed"
      contentId="1.5760"
      contentName="grid"
    />
  );
  expect(snapshot).toMatchSnapshot();
});

it('Twitter user timeline ', () => {
  const snapshot = felaSnapshotter(
    <Twitter
      content="<a class=\&quot;twitter-timeline\&quot; data-partner=\&quot;tweetdeck\&quot; href=\&quot;https://twitter.com/woodenecho/timelines/890172042527592448\&quot;>Testing collections - Curated tweets by woodenecho</a>"
      caption="timeline"
      credit="timeline"
      embedType="user timeline"
      elementType="embedElement"
      inputTemplate="com.polobase.TwitterEmbed"
      contentId="1.5761"
      contentName="timeline"
    />
  );
  expect(snapshot).toMatchSnapshot();
});

it('Twitter search timeline ', () => {
  const snapshot = felaSnapshotter(
    <Twitter
      content="<a class=\&quot;twitter-timeline\&quot; href=\&quot;https://twitter.com/hashtag/webdesign\&quot; data-widget-id=\&quot;890169905349357568\&quot;>#webdesign Tweets</a>"
      caption="search"
      credit="search"
      embedType="search timeline"
      elementType="embedElement"
      inputTemplate="com.polobase.TwitterEmbed"
      contentId="7.4413"
      contentName="search"
    />
  );
  expect(snapshot).toMatchSnapshot();
});

it('Twitter video ', () => {
  const snapshot = felaSnapshotter(
    <Twitter
      content="<blockquote class=\&quot;twitter-video\&quot; data-lang=\&quot;en\&quot;><p lang=\&quot;en\&quot; dir=\&quot;ltr\&quot;>Tips For Combating Climate Change <a href=\&quot;https://t.co/QXm60aJyTR\&quot;>pic.twitter.com/QXm60aJyTR</a></p>&mdash; The Onion (@TheOnion) <a href=\&quot;https://twitter.com/TheOnion/status/890050311196254211\&quot;>July 26, 2017</a></blockquote>"
      caption="twitter"
      credit="twitter"
      embedType="video"
      elementType="embedElement"
      inputTemplate="com.polobase.TwitterEmbed"
      contentId="1.5786"
      contentName="twitter"
    />
  );
  expect(snapshot).toMatchSnapshot();
});
