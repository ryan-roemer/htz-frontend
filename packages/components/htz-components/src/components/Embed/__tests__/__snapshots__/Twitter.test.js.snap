// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Twitter grid  1`] = `
Object {
  "component": <div>
    <div
      className="htz-a htz-b htz-c htz-d htz-e htz-f"
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "<a class=\\\\\\"twitter-grid\\\\\\" data-partner=\\\\\\"tweetdeck\\\\\\" href=\\\\\\"https://twitter.com/woodenecho/timelines/890172042527592448\\\\\\">Testing collections</a>",
          }
        }
      />
    </div>
  </div>,
  "styles": ".htz-a{overflow:auto}.htz-b{height:400px}.htz-c{max-height:800px}.htz-d{position:relative}.htz-e{margin:0 auto}.htz-f .twitter-tweet{margin-bottom:0 !important}",
}
`;

exports[`Twitter search timeline  1`] = `
Object {
  "component": <div>
    <div
      className="htz-a htz-b htz-c htz-d htz-e htz-f"
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "<a class=\\\\\\"twitter-timeline\\\\\\" href=\\\\\\"https://twitter.com/hashtag/webdesign\\\\\\" data-widget-id=\\\\\\"890169905349357568\\\\\\">#webdesign Tweets</a>",
          }
        }
      />
    </div>
  </div>,
  "styles": ".htz-a{overflow:auto}.htz-b{height:auto}.htz-c{max-height:800px}.htz-d{position:relative}.htz-e{margin:0 auto}.htz-f .twitter-tweet{margin-bottom:0 !important}",
}
`;

exports[`Twitter single tweet  1`] = `
Object {
  "component": <div>
    <div
      className="htz-a htz-b htz-c htz-d htz-e htz-f"
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "<blockquote class=\\\\\\"twitter-tweet\\\\\\" data-lang=\\\\\\"en\\\\\\"><p lang=\\\\\\"en\\\\\\" dir=\\\\\\"ltr\\\\\\"><a href=\\\\\\"https://twitter.com/hashtag/Netflix?src=hash\\\\\\">#Netflix</a>'s <a href=\\\\\\"https://twitter.com/hashtag/BoJackHorseman?src=hash\\\\\\">#BoJackHorseman</a> finally has a season 4 premiere date <a href=\\\\\\"https://t.co/xt08ol2mN2\\\\\\">https://t.co/xt08ol2mN2</a> <a href=\\\\\\"https://t.co/IYI8pq05rc\\\\\\">pic.twitter.com/IYI8pq05rc</a></p>— UPROXX (@UPROXX) <a href=\\\\\\"https://twitter.com/UPROXX/status/884416188280864768\\\\\\">July 10, 2017</a></blockquote>",
          }
        }
      />
    </div>
  </div>,
  "styles": ".htz-a{overflow:auto}.htz-b{height:auto}.htz-c{max-height:800px}.htz-d{position:relative}.htz-e{margin:0 auto}.htz-f .twitter-tweet{margin-bottom:0 !important}",
}
`;

exports[`Twitter user timeline  1`] = `
Object {
  "component": <div>
    <div
      className="htz-a htz-b htz-c htz-d htz-e htz-f"
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "<a class=\\\\\\"twitter-timeline\\\\\\" data-partner=\\\\\\"tweetdeck\\\\\\" href=\\\\\\"https://twitter.com/woodenecho/timelines/890172042527592448\\\\\\">Testing collections - Curated tweets by woodenecho</a>",
          }
        }
      />
    </div>
  </div>,
  "styles": ".htz-a{overflow:auto}.htz-b{height:400px}.htz-c{max-height:800px}.htz-d{position:relative}.htz-e{margin:0 auto}.htz-f .twitter-tweet{margin-bottom:0 !important}",
}
`;

exports[`Twitter video  1`] = `
Object {
  "component": <div>
    <div
      className="htz-a htz-b htz-c htz-d htz-e htz-f"
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "<blockquote class=\\\\\\"twitter-video\\\\\\" data-lang=\\\\\\"en\\\\\\"><p lang=\\\\\\"en\\\\\\" dir=\\\\\\"ltr\\\\\\">Tips For Combating Climate Change <a href=\\\\\\"https://t.co/QXm60aJyTR\\\\\\">pic.twitter.com/QXm60aJyTR</a></p>— The Onion (@TheOnion) <a href=\\\\\\"https://twitter.com/TheOnion/status/890050311196254211\\\\\\">July 26, 2017</a></blockquote>",
          }
        }
      />
    </div>
  </div>,
  "styles": ".htz-a{overflow:auto}.htz-b{height:auto}.htz-c{max-height:800px}.htz-d{position:relative}.htz-e{margin:0 auto}.htz-f .twitter-tweet{margin-bottom:0 !important}",
}
`;
